src/views/Room.vue

```diff
@@ -202,6 +203,18 @@ export default {
         },
       })
     },
+    reconnectRtc() {
+      if (this.signalingConnectedBefore) {
+        // TODO: show "Palava server not reachable" or "Network not reachable" overlay
+        if (navigator.online) {
+          rtc.reconnect()
+        } else {
+          window.addEventListener('online',  function() {rtc.reconnect()});
```

Comment:
The event listener needs to be unregistered, when fired, or they will "stack up" with every new error

Reply:
This whole online/offline state thing could also be considered an extra feature (and treated separately from connection errors)

Reply:
On some point we could think of introducing some kind of state machine. Otherwise it will get hard to keep the overview.

Reply:
I think as long as the websocket connection stands, everything is ok - and when it fails, this is only a safeguard not to try it again until it could actually work :-) I don't see a need to treat this separately.

Reply:
I like the idea of having a state-machine (probably even the client library) - which brings together signaling, peer connection, and network/online status